frame/transaction-payment/src/lib.rs

```diff
@@ -214,18 +215,28 @@ impl Default for Releases {
 
 pub trait Trait: frame_system::Trait {
 	/// The currency type in which fees will be paid.
-	type Currency: Currency<Self::AccountId> + Send + Sync;
-
-	/// Handler for the unbalanced reduction when taking transaction fees. This is either one or
-	/// two separate imbalances, the first is the transaction fee paid, the second is the tip paid,
-	/// if any.
-	type OnTransactionPayment: OnUnbalanced<NegativeImbalanceOf<Self>>;
+	type Balance: AtLeast32BitUnsigned
```

Comment:
I feel like this might be very subjective, but I agree that it feels better if it's moved to `OnTransactionPayment`. fixed it in 88a48d06d267f7a04e6b8b6e87df7ad3bedef7cb